File: src/main/scala/info/crlog/higgs/protocols/boson/demo/DemoClient.scala
package info.crlog.higgs.protocols.boson.demo

import info.crlog.higgs.protocols.boson.BosonClient
import com.fillta.higgs.boson.demo.{NestedField, PoloExample}


/**
 * @author Courtney Robinson <courtney@crlog.info>
 */
object DemoClient {
  def main(args: Array[String]) {
    val client = new BosonClient("BosonTest", 8080, "localhost")
    client.connect()
    var count = 0
    val max = readInt()
    val polo = new PoloExample(100)
    for (i <- 1 to max) {
      client.invoke("polo",
        Array(polo),
        (m: PoloExample) => {
          println(m)
          count += 1
          // println(count)
        }, subscribe = false)
      //      client.invoke("all",
      //        Array(1.2F, 140, null, Map("a" -> 120), Array(1, 2, 356), false, "a string",
      //          List("Some List"), polo),
      //        (m: Array[AnyRef]) => {
      //          //          for (i <- m) {
      //          //            println(i)
      //          //          }
      //          count += 1
      //         // println(count)
      //        }, subscribe = false)
    }
  }
}

File: higgs-boson/src/main/java/com/fillta/higgs/boson/demo/DemoClient.java
package com.fillta.higgs.boson.demo;

import com.fillta.functional.Function1;
import com.fillta.higgs.boson.BosonClient;
import com.fillta.higgs.boson.BosonConnectFuture;
import com.fillta.higgs.events.listeners.ChannelEventListener;
import com.google.common.base.Optional;
import io.netty.channel.ChannelHandlerContext;

/**
 * @author Courtney Robinson <courtney@crlog.info>
 */
public class DemoClient {
	public static void main(String... args) {
		BosonClient client = new BosonClient();
		client.onException(new ChannelEventListener() {
			public void triggered(ChannelHandlerContext ctx, Optional<Throwable> ex) {
				ex.get().printStackTrace();
			}
		});

		BosonConnectFuture future = client.connect("BosonDemo", "localhost", 8080, true);
		for (int i = 0; i < 500000; i++) {
			//even if we're not connected the client will buffer messages until we are, then send them
			future.invoke("polo", new Function1<PoloExample>() {
				public void apply(PoloExample a) {
					System.out.println(a.i);
				}
			}, new PoloExample(i));
		}
		System.out.println("Done sending");
	}
}
